products/accredited-representation-management/technical-documentation/appoint-a-representative/poa_request_proposal.md

```diff
@@ -0,0 +1,160 @@
+  end
+
+  def expire!
+    terminate
+    self.status = 'expired'
+
+    self.save
+  end
+
+  private
+
+  def terminate
+    self.first_name = nil
+    self.last_name = nil
+    self.city = nil
+    self.state_code = nil
+    self.zip_code = nil
+    self.consent_data = nil
+  end
+end
+```
+
+#### The Migration
+```ruby
+class CreatePowerOfAttorneyRequests < ActiveRecord::Migration[7.1]
+  def change
+    create_table :power_of_attorney_requests do |t|
+      t.references :user_account, foreign_key: true, type: :uuid
+      t.references :requested_individual, foreign_key: {to_table: 'accredited_individual'}, type: :uuid
+
+      t.string :first_name_bidx
+      t.string :last_name_bidx
+      t.string :city_bidx
+      t.string :state_bidx
+      t.string :zip_code_bidx
+
+      t.text :consent_data_ciphertext
+
+      t.string :status
+      t.datetime :expires_at
+      t.references :accepted_or_declined_by, foreign_key: {to_table: 'accredited_individual'}, type: :uuid
+      t.string :decline_reason
+
+      t.text :encrypted_kms_key
+      t.timestamps
+
+      t.index :first_name_bidx
+      t.index :last_name_bidx
+      t.index :city_bidx
+      t.index :state_bidx
+      t.index :zip_code_bidx
+    end
+  end
+end
+```
+
+#### Consent Data
+Consent data will be treated as jsonb and should only contain boolean values.
+```ruby
+{
+  address_change_authorization: true,
+  limitation_alcohol: true,
+  limitation_hiv: true,
+  limitation_s_c_a: true,
+  limitation_h_i_v: false
+}
+```
+
+### Short term persistence
+Due to the sensitive nature of this data, the various encrypted fields will only be stored for XX days. After XX days or a terminal status is reached (i.e. the request is accepted) only metadata will be kept in the database.
+
+#### Cleanup Job
+This would be a scheduled job and could run on any cadence. The first choice would be once per day to match `InProgressFormCleaner`.
+
+```ruby
+class PowerOfAttorneyRequestCleaner
+  include Sidekiq::Job
+
+  def perform
+    requests = PowerOfAttorneyRequest.where(status: 'pending').where('expires_at < ?', Time.now.utc)
+    logger.info("Deleting #{requests.count} old power of attorney requests")
+    requests.each do |request|
+      request.expire!
+    end
+  end
+end
+```
+
+#### Terminal Actions by Users
+The model helper methods `accept!` and `decline!` should be used when VA.gov users take terminal actions.
+
+## Risks
+- `blind_index` is susceptible to leakage in the same way any deterministic encryption method is.
+- Mixing elevated PII fields (name + location) makes this dataset a more appealing target to attackers.
+- The medical consent limitations may be considered PHI. However, one could argue that limiting access to certain types of records does not necessarily mean a user has those medical conditions.
+
+## Future Considerations
+- Encryption in transit is not considered in this document. That may be an additional security concern for those sending data to VA.gov.
```

Comment:
Do we need anything beyond standard HTTPS/TLS for encryption in transit?

Reply:
I honestly don't know if there are different security needs/requirements for reading this data from the front end due to the resultant sets including many Veteran rows instead of a single Veteran.